Uchu.World/Objects/Components/ReplicaComponents/RailActivatorComponent.cs

```diff
@@ -0,0 +1,245 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using RakDotNet.IO;
+using Uchu.Core;
+using Uchu.World.Client;
+
+namespace Uchu.World
+{
+    public class RailActivatorComponent : ReplicaComponent
+    {
+        public override ComponentId Id => ComponentId.RailActivator;
+
+        // from object settings
+        private int RailActivatorComponentId;
+
+        private bool ActivatorActive;
+
+        private bool ActivatorDamageImmune;
+
+        private string LoopSound;
+
+        private bool NoAggro;
+
+        private bool NotifyActivatorArrived;
+
+        private string Path;
+
+        private bool PathDirection;
+
+        private uint PathStart;
+
+        private bool ShowNameBillboard;
+
+        private bool UseDb;
+
+        // from cdclient
+        private int? StartEffectId;
+        private string StartEffectType;
+
+        private int? DuringEffectId;
+        private string DuringEffectType;
+
+        private int? StopEffectId;
+        private string StopEffectType;
+
+        private string StartSound;
+        private string StopSound;
+
+        private string StartAnim;
+        private string LoopAnim;
+        private string StopAnim;
+
+        private bool CameraLocked;
+        private bool CollisionEnabled;
+
+        protected RailActivatorComponent()
+        {
+            Listen(OnStart, async () =>
+            {
+                RailActivatorComponentId = GameObject.Lot.GetComponentId(ComponentId.RailActivator);
+
+                var railComponent = (await ClientCache.GetTableAsync<Core.Client.RailActivatorComponent>())
+                    .First(c => c.Id == RailActivatorComponentId);
+
+                ActivatorDamageImmune = (bool) GameObject.Settings["rail_activator_damage_immune"];
+                LoopSound = (string) GameObject.Settings["rail_loop_sound"];
+                NoAggro = (bool) GameObject.Settings["rail_no_aggro"];
+                NotifyActivatorArrived = (bool) GameObject.Settings["rail_notify_activator_arrived"];
+                Path = (string) GameObject.Settings["rail_path"];
+                // TODO: investigate why PathDirection 0 makes the movement a straight line instead of the reverse path
```

Comment:
How much was this looked into before? Anything note-worthy for anyone who looks at this later?

Reply:
Been a couple months since I wrote that, IIRC I ended up sending and receiving the exact same packets as I found in the captures from live, but it just didn't work. However I haven't added a handler for `RequestRailActivatorState`, maybe that's related.